botocore/loaders.py

```diff
@@ -446,14 +447,23 @@ def load_data(self, name):
         :type name: str
         :param name: The data path, i.e ``ec2/2015-03-01/service-2``.
 
-        :return: The loaded data.  If no data could be found then
-            a DataNotFoundError is raised.
+        :type return_path: bool
+        :param return_path: Whether to return the path where the data was
+            loaded from as a second return value.
+
+        :return: The loaded data. If return_path is True, the path to the
+            data file is returned as string in the second return value. If
+            no data could be found then a DataNotFoundError is raised.
 
         """
         for possible_path in self._potential_locations(name):
             found = self.file_loader.load_file(possible_path)
             if found is not None:
-                return found
+                if return_path:
+                    return found, possible_path
+                else:
+                    return found
+
```

Comment:
i wonder if it would be better to create a new function that returns both the loaded data and the path rather than changing the data structure of a returnable given a certain condition

Reply:
How about we make a new function that returns both indiscriminately. This function can start calling that new function and only return the data. That will give us the new API surface area, allows us to cache both at the lower function, and keep this method functionally the same.